routes/route.py

```diff
@@ -15,12 +16,19 @@ def resource_not_found(e):
     return jsonify(error=str(e)), 400
 
 
-@routes.route('/test', methods=['GET'])
+@routes.route('/CompareSequence/', methods=['POST'])
 def vernal():
-    print("Executing VERNAL similarity functions")
-    moduleLibraryPath = os.path.join(CURRENT_DIRECTORY, "../core/tools/GraphData/3dMotifAtlas_ALL_one_of_each_graph.cPickle")
-    tempResponsePath = os.path.join(CURRENT_DIRECTORY, "../core/tools/GraphData/response.json") #BP2 output
-    
-    return jsonify(graph_compare.k_most_similar_bp2(moduleLibraryPath, tempResponsePath))
+
+    bp_output = "" 
+    dataset = ""
+    try: 
+        bp_output = eval(request.form.get("graphs"))
+        dataset = eval(request.form.get("dataset", default="ALL", type=str))
+    except Exception as e:
+        abort(400, "Vernal failed to process graph input: " + str(e))
+    print("Executing VERNAL similarity functions with the ", dataset.upper(), " dataset")
+    moduleLibraryPath = os.path.join(CURRENT_DIRECTORY, "../core/tools/GraphData/")
+    #tempResponsePath = os.path.join(CURRENT_DIRECTORY, "../core/tools/GraphData/response.json") #BP2 output
+    return jsonify(graph_compare.k_most_similar_bp2(moduleLibraryPath, bp_output, dataset))
```

Comment:
kinda nitpicking but you could have this in the `try` since it won't get executed anyways if an exception is thrown and it aborts. It'll look cleaner and you won't need to define `bp_output` and `dataset` before the `try:`

Reply:
I'll probably fix this in the next commit I do, thx for check